src/App.tsx

```diff
@@ -1,26 +1,170 @@
+      setErrorMessage(Errors.UnableToDelete);
+      inputAddRef?.current?.focus();
+      throw err;
+    } finally {
+      setLoadingTodoIds(prev => prev.filter(id => id !== todoId));
+    }
+  };
+
+  const onUpdateTodo = async (todoToUpdate: Todo) => {
+    setLoadingTodoIds(prev => [...prev, todoToUpdate.id]);
+    try {
+      const updatedTodo = await todoService.updateTodo(todoToUpdate);
+
+      setTodos(prev =>
+        prev.map(todo => (todo.id === updatedTodo.id ? updatedTodo : todo)),
+      );
+    } catch (err) {
+      setErrorMessage(Errors.UnableToUpdate);
+      throw err;
+    } finally {
+      setLoadingTodoIds(prev => prev.filter(id => id !== todoToUpdate.id));
+    }
+  };
+
+  const onToggleAll = async () => {
+    if (todosActiveNumber > 0) {
+      todos
+        .filter(todo => !todo.completed)
+        .forEach(item => onUpdateTodo({ ...item, completed: true }));
+    } else {
+      todos.forEach(todo => onUpdateTodo({ ...todo, completed: false }));
+    }
+  };
+
+  const onClearCompleted = async () => {
+    const completedTodo = todos.filter(todo => todo.completed);
+
+    completedTodo.forEach(todo => onRemoveTodo(todo.id));
+  };
+
+  const filteredTodos = todos.filter(todo => {
+    switch (filteredField) {
+      case FilterType.Active:
+        return !todo.completed;
+      case FilterType.Completed:
+        return todo.completed;
+      default:
+        return todos;
+    }
+  });
+
+  if (!todoService.USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">
-          React Todo App - Add and Delete
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          setErrorMessage={setErrorMessage}
+          onAddTodo={onAddTodo}
+          isInputDisabled={!!tempTodo}
+          onToggleAll={onToggleAll}
+          todosLength={todos.length}
+          inputRef={inputAddRef}
+          areAllTodosCompleted={areAllTodosCompleted}
+        />
+
+        {(todos.length > 0 || tempTodo) && (
+          <>
+            <TodoList
+              todos={filteredTodos}
+              onRemoveTodo={onRemoveTodo}
+              onUpdateTodo={onUpdateTodo}
+              loadingTodoIds={loadingTodoIds}
+              tempTodo={tempTodo}
+            />
+            <Footer
+              todos={todos}
+              filteredField={filteredField}
+              setFilteredField={setFilteredField}
+              activeTodo={todosActiveNumber}
+              onClearCompleted={onClearCompleted}
+            />
+          </>
+        )}
+      </div>
+      <ErrorNotification
```

Comment:
1) use `!!` to convert value to boolean type
2) don't forget about indents
```suggestion
        {(!!todos.length || tempTodo) && (
          <>
            <TodoList
              todos={filteredTodos}
              onRemoveTodo={onRemoveTodo}
              onUpdateTodo={onUpdateTodo}
              loadingTodoIds={loadingTodoIds}
              tempTodo={tempTodo}
            />
            
            <Footer
              todos={todos}
              filteredField={filteredField}
              setFilteredField={setFilteredField}
              activeTodo={todosActiveNumber}
              onClearCompleted={onClearCompleted}
            />
          </>
        )}
      </div>
      
      <ErrorNotification
```